tests/unit/test_pandas_df_utils.py

```diff
@@ -94,3 +96,40 @@ def test_filter_by():
     assert len(filtered_df) == len(user_df) - 2
     # Check filtered out record
     assert len(filtered_df.loc[(user_df['user_id'] == 1)]) == 0
+
+
+def test_csv_to_libffm():
+    df_feature = pd.DataFrame({
+        'rating': [1, 0, 0, 1, 1],
+        'field1': ['xxx1', 'xxx2', 'xxx3', 'xxx4', 'xxx5'],
+        'field2': ['yyy1', 'yyy2', 'yyy3', 'yyy4', 'yyy5'],
+        'field3': [1.0, 2.0, 3.0, 4.0, 5.0],
+        'field5': [1, 2, 3, 4, 5]
+    })
+
+    import tempfile
+    import os
+
+    filedir = tempfile.tempdir
+    filename = 'test'
+    filepath = os.path.join(filedir, filename)
+
+    df_feature_libffm = df_to_libffm(df_feature, col_rating='rating', filepath=filepath)
+
+    # Check if the dim is the same.
+    assert df_feature_libffm.shape == df_feature.shape
+
+    # Check if the columns are converted successfully.
+    assert df_feature_libffm.iloc[0, :].values.tolist() == [1, '1:1:1', '2:6:1', '3:3:1.0', '4:11:1']
```

Comment:
is this the entire dataframe output? I was expecting 5 values?

Reply:
Ah I just took the first row - yea update to use the entire converted data to test.